File: java/parser/src/main/java/gov/lbl/parser/domain/Short_class_specifier_value.java
package gov.lbl.parser.domain;

public class Short_class_specifier_value {
    private Base_prefix base_prefix;
    private Name name;
    private Array_subscripts array_subscripts;
    private Class_modification class_modification;
    private Comment comment;
    private Enum_list enum_list;

    public Short_class_specifier_value(Base_prefix base_prefix, Name name, Array_subscripts array_subscripts,
            Class_modification class_modification, Comment comment, Enum_list enum_list) {
        this.base_prefix = base_prefix;
        this.name = name;
        this.array_subscripts = array_subscripts;
        this.class_modification = class_modification;
        this.comment = comment;
        this.enum_list = enum_list;
    }
}

File: json2mo/shortClassSpecifierValue.js
function parse (content, rawJson = false) {
  const basePrefixParser = require('./basePrefix')
  const nameParser = require('./name')
  const arraySubscriptsParser = require('./arraySubscripts')
  const classModificationParser = require('./classModification')
  const commentParser = require('./comment')
  const enumListParser = require('./enumList')

  let moOutput = ''

  if (content.name == null) {
    moOutput += 'enumeration ('
    if (content.enum_list != null) {
      moOutput += enumListParser.parse(content.enum_list, rawJson)
    } else {
      moOutput += ':'
    }
    moOutput += ') '
  } else {
    moOutput += basePrefixParser.parse(content.base_prefix, rawJson)
    if (content.name != null) {
      moOutput += nameParser.parse(content.name, rawJson)
    }
    if (content.array_subscripts != null) {
      moOutput += arraySubscriptsParser.parse(content.array_subscripts, rawJson)
    }
    if (content.class_modification != null) {
      moOutput += classModificationParser.parse(content.class_modification, rawJson)
    }
  }
  if (rawJson) {
    if (content.comment != null) {
      moOutput += commentParser.parse(content.comment, rawJson)
    }
  } else {
    if (content.description != null) {
      moOutput += commentParser.parse(content.description, rawJson)
    }
  }

  return moOutput
}

module.exports = { parse }
